lib/client/api.go

```diff
@@ -888,7 +888,12 @@ func (tc *TeleportClient) getTargetNodes(ctx context.Context, proxy *ProxyClient
 			return nil, trace.Wrap(err)
 		}
 		for i := 0; i < len(nodes); i++ {
-			retval = append(retval, nodes[i].GetAddr())
+			addr := nodes[i].GetAddr()
+			if addr == "" {
+				// address is empty, try dialing by UUID instead.
+				addr = fmt.Sprintf("%s:0", nodes[i].GetName())
+			}
```

Comment:
Should this fallback live within `GetAddr()` method itself to prevent future confusion?

Reply:
This value isn't a useful default outside of very specific contexts.  The rest of teleport seems to know that `GetAddr() == ""` means that we're dealing with a reversetunnel node... we just missed these two cases because there wasn't proper test coverage for `tsh` operations against reversetunnel nodes.

I *do* want to do some refactoring around how both `tsh` and the web UI handle dialing nodes, but I don't want that to be a blocker for fixing the issue at hand.